src/Ocelot/RateLimiting/Middleware/RateLimitingMiddleware.cs

```diff
@@ -71,8 +76,13 @@ public async Task Invoke(HttpContext httpContext)
             //set X-Rate-Limit headers for the longest period
             if (!options.DisableRateLimitHeaders)
             {
-                var headers = _limiter.GetHeaders(httpContext, identity, options);
-                httpContext.Response.OnStarting(SetRateLimitHeaders, state: headers);
+                var httpContextAccessor = _container.GetService<IHttpContextAccessor>();
```

Comment:
We should check if the original http context is null - this shouldn't happen though -, why not use
```c#
var originalHttpContext = httpContextAccessor?.HttpContext;
if(originalHttpContext != null)
{
   var headers = _limiter.GetHeaders(originalHttpContext, identity, options);
   originalHttpContext.Response.OnStarting(SetRateLimitHeaders, state: headers);
}
```
instead?

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/1307/commits/42e7703385a74cd20ee30e2861829ea1e4de3750 ✔️